core/src/banking_stage.rs

```diff
@@ -1275,37 +1297,41 @@ impl BankingStage {
     }
 
     /// try to add filtered forwardable and valid packets to forward buffer;
-    /// returns if forward_buffer is still accepting packets, and how many packets added.
+    /// returns if forward buffer is still accepting packets, and vector of
+    /// packet indexes of those were added to forward buffer.
     fn add_filtered_packets_to_forward_buffer(
         forward_buffer: &mut ForwardPacketBatchesByAccounts,
         packets_to_process: &[Arc<ImmutableDeserializedPacket>],
         transactions: &[SanitizedTransaction],
         transaction_to_packet_indexes: &[usize],
-        retained_transaction_indexes: &[usize],
+        forwardable_transaction_indexes: &[usize],
         dropped_tx_before_forwarding_count: &mut usize,
-    ) -> bool {
+    ) -> (bool, Vec<usize>) {
```

Comment:
hmm, seems like maybe we can get rid of the bool return value and just check `accepted_packet_indexes .len == forwardable_transaction_indexes.len()`

Reply:
nice, made change.